src/backend/distributed/operations/stage_protocol.c

```diff
@@ -726,7 +725,7 @@ WorkerCreateShardCommandList(Oid relationId, int shardIndex, uint64 shardId,
 		{
 			referencedShardId = shardId;
 		}
-		else if (PartitionMethod(referencedRelationId) == DISTRIBUTE_BY_NONE)
+		else if (IsCitusTableType(referencedRelationId, REFERENCE_TABLE))
```

Comment:
these are tricky ones :) In theory, we could use `DIST_NONE` instead of reference table, but this code path is not triggered by local tables (we don't support local - distributed foreign keys yet). But, I'm good with the currentt selection